framework/test/jest/specs/integration/in_memory_channel.spec.js

```diff
@@ -0,0 +1,200 @@
+	describe('after registering itself to the bus', () => {
+		let inMemoryChannelAlpha;
+		let inMemoryChannelBeta;
+		let bus;
+
+		beforeEach(async () => {
+			// Arrange
+			bus = new Bus(
+				{
+					wildcard: true,
+					delimiter: ':',
+					maxListeners: 1000,
+				},
+				logger,
+				config
+			);
+
+			inMemoryChannelAlpha = new InMemoryChannel(
+				alpha.moduleAlias,
+				alpha.events,
+				alpha.actions
+			);
+			await inMemoryChannelAlpha.registerToBus(bus);
+
+			inMemoryChannelBeta = new InMemoryChannel(
+				beta.moduleAlias,
+				beta.events,
+				beta.actions
+			);
+			await inMemoryChannelBeta.registerToBus(bus);
+		});
+
+		describe('#subscribe', () => {
+			it('should be able to subscribe to an event.', done => {
+				// Arrange
+				const betaEventData = '#DATA';
+				const eventName = beta.events[0].key();
+
+				// Act
+				inMemoryChannelAlpha.subscribe(
+					`${beta.moduleAlias}:${eventName}`,
+					data => {
+						// Assert
+						expect(Event.deserialize(data).data).toBe(betaEventData);
+						done();
+					}
+				);
+
+				inMemoryChannelBeta.publish(
+					`${beta.moduleAlias}:${eventName}`,
+					betaEventData
+				);
+			});
+
+			it('should be able to subscribe to an event once.', done => {
+				// Arrange
+				const betaEventData = '#DATA';
+				const eventName = beta.events[0].key();
+
+				// Act
+				inMemoryChannelAlpha.once(`${beta.moduleAlias}:${eventName}`, data => {
+					// Assert
+					expect(Event.deserialize(data).data).toBe(betaEventData);
+					done();
+				});
+
+				inMemoryChannelBeta.publish(
+					`${beta.moduleAlias}:${eventName}`,
+					betaEventData
+				);
+			});
+
+			it('should be able to subscribe to an unregistered event.', async done => {
+				// Arrange
+				const omegaEventName = 'omegaEventName';
+				const omegaAlias = 'omegaAlias';
+				const dummyData = '#DATA';
+				const inMemoryChannelOmega = new InMemoryChannel(
+					omegaAlias,
+					[omegaEventName],
+					{}
+				);
+
+				// Act
+				inMemoryChannelAlpha.subscribe(
+					`${omegaAlias}:${omegaEventName}`,
+					data => {
+						// Assert
+						expect(data).toBe(data);
+						done();
+					}
+				);
+
+				await inMemoryChannelOmega.registerToBus(bus);
+
+				inMemoryChannelOmega.publish(
+					`${omegaAlias}:${omegaEventName}`,
+					dummyData
+				);
+			});
+		});
+
+		describe('#publish', () => {
+			it('should be able to publish an event.', done => {
+				// Arrange
+				const alphaEventData = '#DATA';
+				const eventName = alpha.events[0].key();
+
+				// Act
+				inMemoryChannelBeta.once(`${alpha.moduleAlias}:${eventName}`, data => {
+					// Assert
+					expect(Event.deserialize(data).data).toBe(alphaEventData);
+					done();
+				});
+
+				inMemoryChannelAlpha.publish(
+					`${alpha.moduleAlias}:${eventName}`,
+					alphaEventData
+				);
+			});
+		});
+
+		describe('#invoke', () => {
+			it('should be able to invoke its own actions.', async () => {
+				// Act && Assert
+				await expect(
```

Comment:
do we need await here? and below.

Reply:
@shuse2 `invoke` is an async function and to be able to assert it with `resolves` the usage of `await` is needed. See [here](https://jestjs.io/docs/en/tutorial-async#async-await)

Reply:
@shuse2 Problem is assertion is failing but with a Promise :D so if you don't await it you get the following problem.

![image](https://user-images.githubusercontent.com/26697004/55010905-5f827100-4fe5-11e9-8b15-402e5ebe4032.png)